pennylane/math/matrix_manipulation.py

```diff
@@ -433,7 +433,7 @@ def convert_to_su2(U, return_global_phase=False):
     else:
         c_phase = math.cast_like(global_phase, 1j)
```

Comment:
just curious why would `cast_like(*, 1j)` have been useful here?

Reply:
I honestly don't know, and I don't want to touch it. Trying to remove or change anything in regards to type casting here either breaks `torch` or `jax` or `tensorflow`.